src/ImageDetector.ts

```diff
@@ -1,20 +1,40 @@
-import * as exec from 'actions-exec-listener'
-import * as core from '@actions/core'
+import * as exec from "actions-exec-listener";
+import * as core from "@actions/core";
 
 export class ImageDetector {
-  async getExistingImages(): Promise<string[]> {
-    const existingSet = new Set<string>([])
-    const ids = (await exec.exec(`docker image ls -q`, [], { silent: true, listeners: { stderr: console.warn }})).stdoutStr.split(`\n`).filter(id => id !== ``)
-    const repotags = (await exec.exec(`docker`, `image ls --format {{.Repository}}:{{.Tag}} --filter dangling=false`.split(' '), { silent: true, listeners: { stderr: console.warn }})).stdoutStr.split(`\n`).filter(id => id !== ``);
+  async getExistingImages(container: "docker" | "podman"): Promise<string[]> {
+    const existingSet = new Set<string>([]);
+    const ids = (
+      await exec.exec(`${container} image ls -q`, [], {
```

Comment:
Pin

Reply:
What we see is that we have replaced `docker image ls -q` with ``${container} image ls -q`, in which we can specify `${container}` to be either `docker` or `podman`.